firestore/api/SolutionCounter/Program.cs

```diff
@@ -0,0 +1,209 @@
+﻿/*
+ * Copyright (c) 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Google.Cloud.Firestore;
+
+namespace GoogleCloudSamples
+{
+    // Counter is a collection of documents (shards)
+    // to realize counter with high frequency.
+
+    // [START counter_classes]
+
+    // counters/${ID}
+    /// <summary>
+    /// Defines the <see cref="Counter" />
+    /// </summary>
+    [FirestoreData]
+    public class Counter
```

Comment:
I believe this is never store in Firestore, in that case it shouldn't be FirestoreData.
But actually, for the example to make a little more sense, maybe the Counter should be stored in Firestore?